src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -525,18 +527,28 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
       return;
     }
 
-    if (!queue_to_submissions_.contains(queue)) {
-      queue_to_submissions_[queue] = {};
+    if (!submissions_queue_.contains(queue)) {
+      std::priority_queue<QueueSubmission, std::vector<QueueSubmission>,
+                          std::function<bool(QueueSubmission, QueueSubmission)>>
+          priority_queue{[](const QueueSubmission& lhs, const QueueSubmission& rhs) -> bool {
+            return lhs.meta_information.pre_submission_cpu_timestamp >
+                   rhs.meta_information.pre_submission_cpu_timestamp;
+          }};
+      submissions_queue_[queue] = priority_queue;
     }
-    CHECK(queue_to_submissions_.contains(queue));
-    queue_to_submissions_.at(queue).emplace_back(std::move(queue_submission_optional.value()));
+
+    submissions_queue_.at(queue).emplace(std::move(queue_submission_optional.value()));
   }
 
   // This method is responsible for retrieving all the timestamps for the "completed" submissions,
-  // for transforming the information of those submissions (in particlular about the command buffers
+  // for transforming the information of those submissions (in particular about the command buffers
   // and debug markers) into the `GpuQueueSubmission` proto, and for sending it to the
-  // `VulkanLayerProducer`. We consider a submission to be "completed" if its last command buffer
-  // timestamp is ready (see `PullCompletedSubmissions`).
+  // `VulkanLayerProducer`. We consider a submission to be "completed" all timestamps that are
+  // associated with this submission are ready.
```

Comment:
There something missing in this sentence, probably "when".

Reply:
Yes, thanks!